ios/FluentUI.Demo/FluentUI.Demo/Demos/ColorDemoController.swift

```diff
@@ -252,6 +245,6 @@ struct DemoColorSection {
 }
 
 private struct DemoColorTheme {
-    var window: UIWindow
-    var windowType: UIWindow.Type
+    var provider: NSObject
+    var providerType: NSObject.Type
```

Comment:
Hmm... I know this is kind of a carry-over from the older version, but I really don't like the idea of passing an object type around for switching behavior, especially when we also have an instance of the object itself.

Here's an idea: instead of this struct, why don't we define a new enum for the different themes? I'd put it in `ColorDemoThemes.swift` and have individual cases for each of the three themes.

From there, you can also provide custom properties on the enum itself (Swift is cool like that), including a getter that would return the appropriate provider for the given value. Then you could do away with both the struct and the `colorProviderThemes` array, and instead use the fact that it's an interable enum to do indexing and lookup. Something like this:

```swift
enum DemoColorTheme: CaseIterable {
    case `default`
    case green
    case none

    var name: String {
        switch self {
        case .default:
            return "Default"
        case .green:
            return "Green"
        case .none:
            return "None"
        }
    }

    var provider: ColorProviding? {
        switch self {
        case .default:
            return DemoColorDefaultTheme()
        case .green:
            return DemoColorGreenTheme()
        case .none:
            return nil
        }
    }
}
```